src/content/BatchRequestContent.ts

```diff
@@ -0,0 +1,295 @@
+    if (requests.size === 0) {
+      const error = new Error("Empty requests map, Please provide at least one request.");
+      error.name = "Empty Requests Error";
+      throw error;
+    }
+    return isParallel(requests) || isSerial(requests) || isSame(requests);
+  }
+
+  /**
+   * @public
+   * Adds a request to the batch request content
+   * @param {BatchRequestStep} request - The request value
+   * @returns The id of the added request
+   */
+  private addRequest(request: BatchItem): string {
+    const limit = BatchRequestContent.requestLimit;
+    if (request.id === "") {
+      const error = new Error(`Id for a request is empty, Please provide an unique id`);
+      error.name = "Empty Id For Request";
+      throw error;
+    }
+    if (this.requests.size === limit) {
+      const error = new Error(`Maximum requests limit exceeded, Max allowed number of requests are ${limit}`);
+      error.name = "Limit Exceeded Error";
+      throw error;
+    }
+    if (this.requests.has(request.id)) {
+      const error = new Error(`Adding request with duplicate id ${request.id}, Make the id of the requests unique`);
+      error.name = "Duplicate RequestId Error";
+      throw error;
+    }
+    this.requests.set(request.id, request);
+    return request.id;
+  }
+
+  /**
+   * @public
+   * Receives a request information object, converts it and adds it to the batch request execution chain
+   * @param requestInformation
+   */
+  public addBatchRequest(requestInformation: RequestInformation): BatchItem {
+    const batchItem = this.toBatchItem(requestInformation);
+    this.addRequest(batchItem);
+    return batchItem;
+  }
+
+  /**
+   * @private
+   * Converts the request information object to a batch item
+   * @param requestInformation - The request information object
+   */
+  private toBatchItem(requestInformation: RequestInformation): BatchItem {
+    if (requestInformation.pathParameters && requestInformation.pathParameters.baseurl === undefined) {
+      requestInformation.pathParameters.baseurl = this.requestAdapter.baseUrl;
+    }
+
+    let uriString = requestInformation.URL;
+
+    Object.keys(defaultUrlReplacementPairs).forEach(replacementKey => {
+      uriString = uriString.replace(replacementKey, defaultUrlReplacementPairs[replacementKey]);
+    });
+
+    const content = requestInformation.content ? new TextDecoder().decode(requestInformation.content) : undefined;
+    let body: Map<string, any> | undefined;
+    if (content !== undefined) {
+      body = new Map<string, any>(Object.entries(JSON.parse(content) as { [s: string]: any }));
+    }
+
+    let headers: Record<string, any> | undefined;
+    if (headers !== undefined) {
+      headers = Object.fromEntries(requestInformation.headers.entries()) as unknown as Record<string, string>;
+    }
+
+    const url = uriString.replace(this.requestAdapter.baseUrl, "");
+
+    const method = requestInformation.httpMethod?.toString();
+
+    return {
+      id: createGuid(),
```

Comment:
The user should still have the ability to pass in their own id, incase they want to use it retreive the response.